### PR TITLE
TEST-#2073: Check that read_csv can use a parse_dates dict.

### LINKED ISSUE #2073: read_csv with Ray engine fails with parse_dates parameter passed as dict
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 18.04
- **Modin version** (`modin.__version__`): 0.7.3+350.g2ca3f34
- **Python version**: 3.8.5
- **Code we can use to reproduce**:
```
import os
os.environ["MODIN_ENGINE"] = "ray"

import pandas
import modin.pandas as pd

test_filename = "test.csv"
csv_single_col = """col1,col2
2000-01-01,2000-01-02
2000-01-03,2000-01-04
"""

try :
    with open(test_filename, "w") as f:
        f.write(csv_single_col)

    df_pandas = pandas.read_csv(test_filename, parse_dates={"foo": ["col1", "col2"]})
    print(df_pandas)
    df_pd = pd.read_csv(test_filename, parse_dates={"foo": ["col1", "col2"]})
    repr(df_pd)
    print(df_pd)
finally:
    os.remove(test_filename)
```

### Source code / logs

```
                         foo
0  2000-01-01 20:00:00-02:00
1  2000-01-03 20:00:00-04:00
Traceback (most recent call last):
  File "test_read_csv.py", line 20, in <module>
    repr(df_pd)
  File "/localdisk/amyskov/modin2/modin/pandas/dataframe.py", line 169, in __repr__
    result = repr(self._build_repr_df(num_rows, num_cols))
  File "/localdisk/amyskov/modin2/modin/pandas/base.py", line 108, in _build_repr_df
    return self.iloc[indexer]._query_compiler.to_pandas()
  File "/localdisk/amyskov/modin2/modin/pandas/indexing.py", line 377, in __getitem__
    result = super(_iLocIndexer, self).__getitem__(row_lookup, col_lookup, ndim)
  File "/localdisk/amyskov/modin2/modin/pandas/indexing.py", line 129, in __getitem__
    qc_view = self.qc.view(row_lookup, col_lookup)
  File "/localdisk/amyskov/modin2/modin/backends/pandas/query_compiler.py", line 2470, in view
    self._modin_frame.mask(row_numeric_idx=index, col_numeric_idx=columns)
  File "/localdisk/amyskov/modin2/modin/engines/base/frame/data.py", line 472, in mask
    return self.copy()
  File "/localdisk/amyskov/modin2/modin/engines/base/frame/data.py", line 643, in copy
    return self.__constructor__(
  File "/localdisk/amyskov/modin2/modin/engines/base/frame/data.py", line 73, in __init__
    ErrorMessage.catch_bugs_and_request_email(
  File "/localdisk/amyskov/modin2/modin/error_message.py", line 51, in catch_bugs_and_request_email
    raise Exception(
Exception: Internal Error. Please email bug_reports@modin.org with the traceback and command that caused this error.
Column widths: 2 != 1
```

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

Check that read_csv can use a parse_dates dict.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2073
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by naren-ponder — APPROVED** [review]

**Review by RehanSD — DISMISSED** [review]
LGTM!

**Review by jeffreykennethli — APPROVED** [review]

File: modin/pandas/test/utils.py [hunk 1]
```diff
@@ -132,6 +132,7 @@
     "list_of_list_of_ints": [[1, 2, 3]],
     "list_of_list_of_strings_and_ints": [["year", 2, "date"]],
     "empty_list": [],
+    "dict": {"year_and_month": [1, 2], "day": ["date"]},
```
[the remark this answers is not in the document]
Comment (mvashishtha) [reply modin/pandas/test/utils.py:135]: I think we've caught all of those now :)

**Comment by mvashishtha** [comment]
> @mvashishtha could you rebase and see if this error remains

@anmyachev I still get a failure locally.

<details>
<summary>stack trace</summary>

```python-traceback
_______________________________________________ TestCsvGlob.test_read_single_csv_with_parse_dates[dict] ________________________________________________

self = <modin.experimental.pandas.test.test_io_exp.TestCsvGlob object at 0x7f934817fa60>, parse_dates = {'day': ['date'], 'year_and_month': [1, 2]}

    @pytest.mark.parametrize(
        "parse_dates",
        [pytest.param(value, id=id) for id, value in parse_dates_values_by_id.items()],
    )
    def test_read_single_csv_with_parse_dates(self, parse_dates):
        try:
            pandas_df = pandas.read_csv(time_parsing_csv_path, parse_dates=parse_dates)
        except Exception as pandas_exception:
            with pytest.raises(Exception) as modin_exception:
                modin_df = pd.read_csv_glob(
                    time_parsing_csv_path, parse_dates=parse_dates
                )
                # Call __repr__ on the modin df to force it to materialize.
                repr(modin_df)
            assert isinstance(
                modin_exception.value, type(pandas_exception)
            ), "Got Modin Exception type {}, but pandas Exception type {} was expected".format(
                type(modin_exception.value), type(pandas_exception)
            )
        else:
>           modin_df = pd.read_csv_glob(time_parsing_csv_path, parse_dates=parse_dates)

modin/experimental/pandas/test/test_io_exp.py:166:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
modin/experimental/pandas/io.py:231: in parser_func
    return _read(**kwargs)
modin/experimental/pandas/io.py:278: in _read
    pd_obj = FactoryDispatcher.read_csv_glob(**kwargs)
modin/core/execution/dispatching/factories/dispatcher.py:188: in read_csv_glob
    return cls.__factory._read_csv_glob(**kwargs)
modin/core/execution/dispatching/factories/factories.py:524: in _read_csv_glob
    return cls.io_cls.read_csv_glob(**kwargs)
modin/logging/logger_decorator.py:128: in run_and_log
    return obj(*args, **kwargs)
modin/core/io/text/csv_glob_dispatcher.py:291: in _read
    column_names = column_names.drop(group).insert(0, new_col_name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Index(['timestamp', 'year', 'month', 'date', 'symbol', 'high', 'low', 'open',
       'close', 'spread', 'volume'],
      dtype='object')
labels = array([1, 2], dtype=object), errors = 'raise'

    def drop(
        self,
        labels: Index | np.ndarray | Iterable[Hashable],
        errors: IgnoreRaise = "raise",
    ) -> Index:
        """
        Make new Index with passed list of labels deleted.

        Parameters
        ----------
        labels : array-like or scalar
        errors : {'ignore', 'raise'}, default 'raise'
            If 'ignore', suppress error and existing labels are dropped.

        Returns
        -------
        dropped : Index
            Will be same type as self, except for RangeIndex.

        Raises
        ------
        KeyError
            If not all of the labels are found in the selected axis
        """
        if not isinstance(labels, Index):
            # avoid materializing e.g. RangeIndex
            arr_dtype = "object" if self.dtype == "object" else None
            labels = com.index_labels_to_array(labels, dtype=arr_dtype)

        indexer = self.get_indexer_for(labels)
        mask = indexer == -1
        if mask.any():
            if errors != "ignore":
>               raise KeyError(f"{list(labels[mask])} not found in axis")
E               KeyError: '[1, 2] not found in axis'

../../opt/anaconda3/envs/modin-dev-py38/lib/python3.8/site-packages/pandas/core/indexes/base.py:6977: KeyError
```

</details>

**Comment by anmyachev** [comment]
> 

@mvashishtha let's skip dict case for read_csv_glob and create separate separate issue for that

**Comment by anmyachev** [comment]
@mvashishtha https://github.com/modin-project/modin/issues/5326

**Review by anmyachev — APPROVED** [review]
LGTM!

**Pull request merged**